### PR TITLE
[MIG] sale_exception: Migration to 16.0

### PR DESCRIPTION
**Pull request by matiasperalta1**

### CODE REVIEW

File: sale_exception/models/sale.py
Comment (ghost) [inline sale_exception/models/sale.py]:
```suggestion
    @api.model_create_multi
```

Keep in mind that you should also adapt the method _check_sale_check_exception. It will receive a list of dict from the create method and only an dict from the write method.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by aliciagaarzo — APPROVED** [review]

**Review by aliciagaarzo — COMMENTED** [review]
After testing the migrated module, I’ve found some improvements:
- Description field is noy properly displayed; user needs to scroll to the right to see it.
- Allow to ignore exceptions even if the “is blocking” option is checked

File: sale_exception/models/sale.py
Comment (sebastienbeau) [inline sale_exception/models/sale.py]: Hi thanks for the migration.

It's not related to your change but after think twice
I think that the check on create is useless !

Indeed if you look at the method sale_check_exception, it explicitly skip all sale order that are not in the state "sale", when we create a sale order the state is always "draft".

So indeed of adding some extra complexity to the code, maybe it's a good opportunity to simplify it and just to remove the inherit on "def create".

I should have seen it before but I miss to review this PR : https://github.com/OCA/sale-workflow/pull/916

@simahawk @rousseldenis @sbejaoui what do you think ?

Comment (sebastienbeau) [reply sale_exception/models/sale.py]: @florian-dacosta

**Comment by sebastienbeau** [comment]
also, can you rebase as base_exception have been merged ?

**Comment by sebastienbeau** [comment]
Note: I also realize that the migration in V9 have drop the commit historic (it was correct in 8 see here: https://github.com/OCA/sale-workflow/commits/8.0?after=605d0ae2da6d37e9acd8466eb77d2cc023d2f289+69&branch=8.0&path%5B%5D=sale_exceptions&qualified_name=refs%2Fheads%2F8.0)

When this PR will be really, I will, before merging the PR, restore the commit history, so please ping me when it's ready, do not waist your time with this point, I will do it.

File: sale_exception/models/sale.py
Comment (florian-dacosta) [reply sale_exception/models/sale.py]: I agree the create override can be removed as it seems useless.

**Comment by rousseldenis** [comment]
> Note: I also realize that the migration in V9 have drop the commit historic (it was correct in 8 see here: https://github.com/OCA/sale-workflow/commits/8.0?after=605d0ae2da6d37e9acd8466eb77d2cc023d2f289+69&branch=8.0&path%5B%5D=sale_exceptions&qualified_name=refs%2Fheads%2F8.0)


@sebastienbeau That's normal as module has been renamed at that time.

File: sale_exception/models/sale.py
Comment (rousseldenis) [inline sale_exception/models/sale.py:9]: Please put models in separate files

**Review by rousseldenis — APPROVED** [review]
LGTM. Minor change

File: sale_exception/tests/test_multi_records.py
Comment (rousseldenis) [inline sale_exception/tests/test_multi_records.py:3]:
```suggestion
from unittest import mock
```

**Review by lmignon — CHANGES_REQUESTED** [review]
LGTM once the requested changes will be implemented

**Comment by rousseldenis** [comment]
@matiasperalta1

**Comment by lmignon** [comment]
@matiasperalta1 Do you plan to work on this PR soon or do you prefer we finalize your work? We need to move forward with this one....

**Comment by sbejaoui** [comment]
I pushed a new PR (#2344) to finalize this one because we need it for one of our projects.

**Comment by rousseldenis** [comment]
@matiasperalta1 Could you close this in favor of #2344 ?

**Pull request merged**
